core/devdoc/outprocess_hld.md

```diff
@@ -0,0 +1,413 @@
+
+This is a brief description of the repsonsibilies of the proxy module.
+
+- **Module_ParseConfigurationFromJson** - The proxy module will serialize the arguments into a JSON string.
+
+- **Module_FreeConfiguration** - The proxy module will free the serialized string.
+
+- **Module_Create** - The proxy Module will format a *Create Module* request and send it via the control channel to the outprocess module, it will wait for a response before returning.
+
+- **Module_Destroy** - The proxy Module will format a *Destroy Module* message and send it via the control channel to the outprocess module. It will not wait for a response.
+
+- **Module_Receive** - The proxy module will serialize the message and send it via the message channel to the outprocess module.
+
+- **Module_Start** - The proxy module will format a *Start Module* message and send it via the control channel to the outprocess module.
+
+- The proxy module will also create a task to read any gateway messages published from the remote module.
+
+
+### Outprocess Module Loader
+
+The proxy module will be loaded by the "outprocess" module loader type. This will be responsible for providing the proxy module API to the gateway.
+
+There is no configuration for the proxy module loader.
+
+The proxy module connection information is placed in the "entrypoint" for the loader.
+
+If the activation type is "launch", then a launch object must be added as a sister element to "activation.type" (example below).
+
+An example of the "loaders" entry:
+
+```JSON
+"loaders" : [
+    {
+        "name" : "myRemoteModuleLoader",
+        "type" : "outprocess"
+    }
+]
+```
+
+An example of proxy module configuration with activation.type "none" :
+
+```JSON
+"modules" : [
+{
+    "name" : "myRemoteModule",
+    "loader" : {
+        "name" : "myRemoteModuleLoader",
+        "type" : "outprocess",
+        "entrypoint" : {
+            "activation.type" : "none",
+            "control.id" : "outprocess_module_control"
+        }
+    }
+    "args" : {...}
+}
+]
+```
+
+An example of proxy module configuration with activation.type "launch" :
+
+```JSON
+"modules" : [
+{
+    "name" : "myRemoteModule",
+    "loader" : {
+        "name" : "myRemoteModuleLoader",
+        "type" : "outprocess",
+        "entrypoint" : {
+            "activation.type" : "launch",
+            "control.id" : "outprocess_module_control",
+            "launch" : {
+                "path" : "/path/to/remote/module.exe",
+                "args" : [
+                    "<available to remote module as argv[1]>",
+                    "<available to remote module as argv[2]>",
+                    "..." ,
+                    "<available to remote module as argv[n]>"
+                ],
+                "grace.period.s" : 2.25
```

Comment:
I would simply this to an integral # of seconds, or an integral # of milliseconds, rather than dealing with a float.